src/components/Form.js

```diff
@@ -60,16 +64,16 @@ export default class Form extends Component {
   }
 
   onChange = (formData, options={validate: false}) => {
-    const liveValidate = this.props.liveValidate || options.validate;
-    const errors = liveValidate ? this.validate(formData) :
-                                  this.state.errors;
-    const errorSchema = toErrorSchema(errors);
-    setState(this, {
-      status: "editing",
-      formData,
-      errors,
-      errorSchema
-    }, () => {
+    const mustValidate = this.props.liveValidate || options.validate;
+    let state;
+    if (!mustValidate) {
+      state = {status: "editing", formData};
+    } else {
+      const errors = this.validate(formData);
+      const errorSchema = toErrorSchema(errors);
+      state = {status: "editing", formData, errors, errorSchema};
+    }
+    setState(this, state, () => {
```

Comment:
Could you please elaborate on this change? As far as I can tell it's just a refactor that doesn't change the semantics of the code. Is that right, or does this somehow change error handling?